File: sdk/java/src/main/java/com/mockforge/sdk/VerificationCount.java
package com.mockforge.sdk;

import java.util.HashMap;
import java.util.Map;

/**
 * Count assertion for verification
 */
public class VerificationCount {
    /**
     * Request must be made exactly N times
     */
    public static Map<String, Object> exactly(int n) {
        Map<String, Object> result = new HashMap<>();
        result.put("type", "exactly");
        result.put("value", n);
        return result;
    }

    /**
     * Request must be made at least N times
     */
    public static Map<String, Object> atLeast(int n) {
        Map<String, Object> result = new HashMap<>();
        result.put("type", "at_least");
        result.put("value", n);
        return result;
    }

    /**
     * Request must be made at most N times
     */
    public static Map<String, Object> atMost(int n) {
        Map<String, Object> result = new HashMap<>();
        result.put("type", "at_most");
        result.put("value", n);
        return result;
    }

    /**
     * Request must never be made (count must be 0)
     */
    public static Map<String, Object> never() {
        Map<String, Object> result = new HashMap<>();
        result.put("type", "never");
        return result;
    }

    /**
     * Request must be made at least once (count >= 1)
     */
    public static Map<String, Object> atLeastOnce() {
        Map<String, Object> result = new HashMap<>();
        result.put("type", "at_least_once");
        return result;
    }
}

File: sdk/dotnet/MockForge.Sdk/VerificationCount.cs
namespace MockForge.Sdk;

/// <summary>
/// Count assertion for verification
/// </summary>
public class VerificationCount
{
    /// <summary>
    /// Request must be made exactly N times
    /// </summary>
    public static Dictionary<string, object> Exactly(int n)
    {
        return new Dictionary<string, object>
        {
            { "type", "exactly" },
            { "value", n }
        };
    }

    /// <summary>
    /// Request must be made at least N times
    /// </summary>
    public static Dictionary<string, object> AtLeast(int n)
    {
        return new Dictionary<string, object>
        {
            { "type", "at_least" },
            { "value", n }
        };
    }

    /// <summary>
    /// Request must be made at most N times
    /// </summary>
    public static Dictionary<string, object> AtMost(int n)
    {
        return new Dictionary<string, object>
        {
            { "type", "at_most" },
            { "value", n }
        };
    }

    /// <summary>
    /// Request must never be made (count must be 0)
    /// </summary>
    public static Dictionary<string, object> Never()
    {
        return new Dictionary<string, object>
        {
            { "type", "never" }
        };
    }

    /// <summary>
    /// Request must be made at least once (count >= 1)
    /// </summary>
    public static Dictionary<string, object> AtLeastOnce()
    {
        return new Dictionary<string, object>
        {
            { "type", "at_least_once" }
        };
    }
}
